src/probnum/filtsmooth/statespace/sde.py

```diff
@@ -82,64 +124,120 @@ class LinearSDE(SDE):
 
     def __init__(
         self,
+        dimension,
         driftmatfun: Callable[[FloatArgType], np.ndarray],
         forcevecfun: Callable[[FloatArgType], np.ndarray],
         dispmatfun: Callable[[FloatArgType], np.ndarray],
+        mde_atol=1e-6,
+        mde_rtol=1e-6,
+        mde_solver="LSODA",
     ):
+
+        # Once different filtering and smoothing algorithms are available,
+        # replicate the scheme from DiscreteGaussian here, in which
+        # the initialisation decides between, e.g., classic and sqrt implementations.
+
         self.driftmatfun = driftmatfun
         self.forcevecfun = forcevecfun
         super().__init__(
-            driftfun=(lambda t, x: driftmatfun(t) @ x + forcevecfun(t)),
+            dimension=dimension,
+            driftfun=(lambda t, x: self.driftmatfun(t) @ x + self.forcevecfun(t)),
             dispmatfun=dispmatfun,
-            jacobfun=(lambda t, x: driftmatfun(t)),
+            jacobfun=(lambda t, x: self.driftmatfun(t)),
         )
 
-    def transition_realization(
+        self.mde_atol = mde_atol
+        self.mde_rtol = mde_rtol
+        self.mde_solver = mde_solver
+
+    def forward_rv(
         self,
-        real,
-        start,
-        stop,
-        step,
+        rv,
+        t,
+        dt=None,
+        _compute_gain=False,
         _diffusion=1.0,
         **kwargs,
     ):
 
-        rv = pnrv.Normal(real, 0 * np.eye(len(real)))
-        return linear_sde_statistics(
-            rv,
-            start,
-            stop,
-            step,
-            self.driftfun,
-            self.driftmatfun,
-            self.dispmatfun,
-            _diffusion=_diffusion,
-        )
+        return self._solve_mde_forward(rv, t, dt, _diffusion=_diffusion)
 
-    def transition_rv(self, rv, start, stop, step, _diffusion=1.0, **kwargs):
+    def backward_rv(
+        self,
+        rv_obtained,
+        rv,
+        rv_forwarded=None,
+        gain=None,
+        t=None,
+        dt=None,
+        _diffusion=1.0,
+        **kwargs,
+    ):
+        raise NotImplementedError("Not available (yet).")
 
-        if not isinstance(rv, pnrv.Normal):
-            errormsg = (
-                "Closed form transitions in linear SDE models is only "
-                "available for Gaussian initial conditions."
-            )
-            raise TypeError(errormsg)
-        return linear_sde_statistics(
+    # Forward (and soon, backward) implementation(s)
+
+    def _solve_mde_forward(self, rv, t, dt, _diffusion=1.0):
+        """Solve forward moment differential equations (MDEs)."""
+        mde, y0 = self._setup_vectorized_mde_forward(
             rv,
-            start,
-            stop,
-            step,
-            self.driftfun,
-            self.driftmatfun,
-            self.dispmatfun,
             _diffusion=_diffusion,
         )
+        sol = scipy.integrate.solve_ivp(
+            mde,
+            (t, t + dt),
```

Comment:
What happens here if `dt = None` ? (Which is the default)

Reply:
Good question. I changed the code to raise a `ValueError` in this case.